### DISCUSSION TITLE
[BUG] Alert Flood

### DISCUSSION THREAD

**Discussion by iurkenty** [nextgenhealthcare/connect#5682] [General]
Description:
I have set up an alert for a Destination Connector whenever I get an ERROR.
I have one message stuck in a queue(intentionally) and I get a flood of alerts for the same message.

**Expected behavior**
I would like to get one alert per message-id per action(ERROR/SENT)

**Actual behavior**
A flood of alerts for the same message with the same ERROR

**Screenshots**
![image](https://user-images.githubusercontent.com/105752711/217597645-a1e00b1c-4ac7-40e4-99d6-a2f601129188.png)

**Environment (please complete the following information):**
 - Ubuntu 22.04
 - Java Distribution/Version [e.g. OpenJDK 11, Java 8 (201)]
 - Mirth 4.2.0

**Comment by iurkenty** [comment]
Is this because the queue is trying to reprocess the message?
If so, is there a work around to only get the alert once?

**Comment by pladesma** [comment]
Yes, the message is reprocessed from the queue, so that's why it errors again and you get another alert. I don't have a good idea for a workaround, but I'll leave this open in case other users want to comment.

**Comment by jonbartels** [comment]
I have a teammate working on a similar problem. We have an HTTP sender that will error because the customer is doing maintenance then succeed later after a few retries. We get an alert for each retry instead of just on the last error.

He does not have any results yet but he is testing a responseTransformer that will:
1. If a message is errored with a "real" error like an HTTP 404, HTTP 500, or a NACK then the message is really errored
2. If a message is errored with an error to retry like a connection timeout or authentication failure then force `responseStatus = QUEUED`
3. Check the send attempts on the message if it exceeds a certain amount, then force the message to error.

I will do my best to report back if our solution works or not.

**Reply by iurkenty** [reply]
Thanks @jonbartels !
